### ISSUE TITLE
MySQLdb._exceptions.ProgrammingError: execute() first

### ISSUE THREAD

**Issue by Marow** [Madpilot0/FlaskDashboard#2]
I get that error when try to login...

`MySQLdb._exceptions.ProgrammingError: execute() first`

**Comment by vlori2k** [comment]
I get the same.. somebody could please help ?

**Comment by SanderWegter** [comment]
Hi

Apologies for the lack of response.

Have you tried to edit the following bit:

https://github.com/Madpilot0/FlaskDashboard/blob/master/lib/Database.py#L32

And add prints for the sql and arguments to determine where the sql error lies. 

```
			except MySQLdb.Error as e:
				print(e)
                                print(f"Query: {sql}")
                                print(f"Args: {str(args)}")
```

I'm currently not actively maintaining this project but in the future I'll probably make an updated version with better error handling.

**Comment by vlori2k** [comment]
I tried that.. still get the exact same error tho ..

**Comment by SanderWegter** [comment]
Yes, but i was curious about the output ;-) Printing the messages won't fix the issue. I'm trying to see if I could quickly help you.

**Comment by vlori2k** [comment]
Seems that the error comes from the database/SQL side:

`(1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups G ON G.id = M.gid WHERE user = 'admin'' at line 1")
Query: SELECT pass,firstname,lastname, G.name FROM users LEFT JOIN groupmembers M ON M.uid = id LEFT JOIN groups G ON G.id = M.gid WHERE user = %s
Args: ['admin']
[2021-02-04 11:24:01,878] ERROR in app: Exception on /login [POST]`

**Comment by SanderWegter** [comment]
Ah, my best guess is that at the time of writing this I used MySQL 5.something. The latest version is (I think) 8.X which has some more reserved words. 'Groups' is one of those.
